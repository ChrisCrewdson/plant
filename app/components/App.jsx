const React = require('react');
const actions = require('../actions');
const PropTypes = require('prop-types');

class App extends React.Component {
  static contextTypes = {
    store: PropTypes.object.isRequired,
  };

  componentWillMount() {
    const { store } = this.context;
    const users = store.getState().get('users');
    if (!users || users.size === 0) {
      store.dispatch(actions.loadUsersRequest());
    }

    const locations = store.getState().get('locations');
    if (!locations || locations.size === 0) {
      store.dispatch(actions.loadLocationsRequest());
    }

    // const user = store.getState().get('user');
    // if(user && user.get('isLoggedIn', false)) {
    //   const locationIds = user.get('locationIds');
    //   if(!locationIds) {
    //     const userId = user.get('_id');
    //     store.dispatch(actions.loadLocationsRequest(userId));
    //   }
    // }
  }

  render() {
    return (
      <div className="react-root">
        {this.props.children}
      </div>
    );
  }
}

App.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  children: PropTypes.object.isRequired,
};

module.exports = App;
