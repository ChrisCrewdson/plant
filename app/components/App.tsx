import React from 'react';
import PropTypes from 'prop-types';
import { actionFunc } from '../actions';

export default class App extends React.Component {
  // TODO: When tsc 3.7+ is in use remove the ! to see hint text on how to change this.
  context!: PlantContext;

  static propTypes = {
    // eslint-disable-next-line react/forbid-prop-types
    children: PropTypes.object.isRequired,
  };

  static contextTypes = {
    // eslint-disable-next-line react/forbid-prop-types
    store: PropTypes.object.isRequired,
  };

  // eslint-disable-next-line camelcase
  UNSAFE_componentWillMount() {
    const { store } = this.context;
    const { users = {}, locations = {} } = store.getState();

    // TODO: This will cause a problem for a non-initialized site
    // that has zero users and zero locations as these values
    // will be 0 which is falsy.
    const usersCount = Object.keys(users).length;
    const locationsCount = Object.keys(locations).length;

    if (!usersCount) {
      store.dispatch(actionFunc.loadUsersRequest());
    }

    if (!locationsCount) {
      store.dispatch(actionFunc.loadLocationsRequest());
    }
  }

  render() {
    const { children } = this.props;
    return (
      <div className="react-root">
        {children}
      </div>
    );
  }
}
