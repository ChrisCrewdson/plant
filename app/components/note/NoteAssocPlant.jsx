
const actions = require('../../actions');
const React = require('react');
const FloatingActionButton = require('material-ui/FloatingActionButton').default;
const InputCombo = require('../common/InputCombo');
const ArrowLeft = require('material-ui/svg-icons/hardware/keyboard-arrow-left').default;
const ArrowRight = require('material-ui/svg-icons/hardware/keyboard-arrow-right').default;
const Errors = require('../common/Errors');
const utils = require('../../libs/utils');
const PropTypes = require('prop-types');
const NoteAssocPlantToggleButton = require('./NoteAssocPlantToggleButton');

class NoteAssocPlant extends React.Component {
  constructor() {
    super();
    this.state = {
      expanded: false,
      filter: '',
    };
    this.expand = this.expand.bind(this);
    this.toggle = this.toggle.bind(this);
  }

  toggle(plantId) {
    const plantIds = this.props.plantIds.indexOf(plantId) >= 0
      ? this.props.plantIds.filter(pId => pId !== plantId)
      : this.props.plantIds.concat(plantId);

    this.props.dispatch(actions.editNoteChange({ plantIds }));
  }

  expand() {
    const expanded = !this.state.expanded;
    this.setState({ expanded });
  }

  renderPlantButton(plant, primary) {
    const _id = plant.get('_id');
    const title = plant.get('title');
    const secondary = !primary && !!plant.get('isTerminated');
    return (<NoteAssocPlantToggleButton
      _id={_id}
      label={title}
      style={{ margin: 12 }}
      toggleFunc={this.toggle}
      primary={primary}
      secondary={secondary}
    />);
  }

  renderPlantButtons(plantIds, plants, selected) {
    return plantIds.map((plantId) => {
      const plant = plants.get(plantId);
      if (!plant) {
        // console.warn(`Missing plant for plantId ${plantId}`);
        return null;
      }
      return this.renderPlantButton(plant, selected);
    });
  }

  render() {
    const { expanded, filter } = this.state;
    const { plantIds, plants } = this.props;

    const checkedPlantIds = utils.filterSortPlants(plantIds, plants, filter);
    const checkedPlants = this.renderPlantButtons(checkedPlantIds, plants, true);

    const uncheckedIds = plants.filter((plant, _id) =>
      plantIds.indexOf(_id) === -1).keySeq().toArray();
    const uncheckedPlantIds = utils.filterSortPlants(uncheckedIds, plants, filter);

    const uncheckedPlants = expanded
      ? this.renderPlantButtons(uncheckedPlantIds, plants, false)
      : null;

    const arrow = (
      <FloatingActionButton
        mini
        onClick={this.expand}
        secondary
        title="Expand/Collapse Unchecked Plants"
      >
        {expanded
          ? <ArrowLeft />
          : <ArrowRight />
        }
      </FloatingActionButton>);

    const filterInput = (<InputCombo
      changeHandler={e => this.setState({ filter: e.target.value.toLowerCase() })}
      label="Filter"
      placeholder="Filter..."
      value={filter}
      name="filter"
    />);

    const errors = this.props.error ? [this.props.error] : [];

    return (
      <div style={{ textAlign: 'left' }}>
        <Errors errors={errors} />
        <div>
          {'Associated plants:'}
          {filterInput}
          {checkedPlants}
          {uncheckedPlants}
          {arrow}
        </div>
      </div>
    );
  }
}

NoteAssocPlant.propTypes = {
  dispatch: PropTypes.func.isRequired,
  error: PropTypes.string,
  plantIds: PropTypes.arrayOf(PropTypes.string).isRequired,
  plants: PropTypes.shape({
    get: PropTypes.func.isRequired,
    filter: PropTypes.func.isRequired,
  }).isRequired,
};

NoteAssocPlant.defaultProps = {
  error: '',
};

module.exports = NoteAssocPlant;
