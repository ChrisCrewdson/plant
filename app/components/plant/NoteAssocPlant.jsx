
const actions = require('../../actions');
const React = require('react');
const FloatingActionButton = require('material-ui/FloatingActionButton').default;
const RaisedButton = require('material-ui/RaisedButton').default;
const ArrowLeft = require('material-ui/svg-icons/hardware/keyboard-arrow-left').default;
const ArrowRight = require('material-ui/svg-icons/hardware/keyboard-arrow-right').default;

class NoteAssocPlant extends React.Component {
  constructor() {
    super();
    this.state = {
      expanded: false
    };
    this.toggle = this.toggle.bind(this);
    this.expand = this.expand.bind(this);
  }

  toggle(plantId) {
    const plantIds = this.props.plantIds.indexOf(plantId) >= 0
      ? this.props.plantIds.filter(pId => pId !== plantId)
      : this.props.plantIds.concat(plantId);

    this.props.dispatch(actions.editNoteChange({plantIds}));
  }

  expand() {
    const expanded = !this.state.expanded;
    this.setState({expanded});
  }

  render() {
    const {expanded} = this.state;
    const {plantIds, plants} = this.props;

    const checkedPlants = plantIds.map(plantId => {
      const plant = plants.get(plantId);
      if(!plant) {
        console.warn(`Missing plant for plantId ${plantId}`);
        return null;
      }

      return <RaisedButton
        key={plantId}
        label={plant.get('title')}
        primary={true}
      />;
    });

    const uncheckedPlants = expanded
      ? plants.reduce((acc, plant) => {
        if(plantIds.indexOf(plant.get('_id') >= 0)) {
          acc.push(<RaisedButton
            key={plant.get('_id')}
            label={plant.get('title')}
          />);
          return acc;
        }
      }, [])
      : null;

    const arrow = <FloatingActionButton
      onClick={this.expand}
      secondary={true}
      title='Expand/Collapse Unchecked Plants'
    >
      {expanded
        ? <ArrowLeft />
        : <ArrowRight />
      }
    </FloatingActionButton>;


    return (
      <div>
        {'Associated plants:'}
        {checkedPlants}
        {uncheckedPlants}
        {arrow}
      </div>
    );
  }
}

NoteAssocPlant.propTypes = {
  dispatch: React.PropTypes.func.isRequired,
  plantIds: React.PropTypes.array.isRequired,
  plants: React.PropTypes.object.isRequired, // Immutable.js Map
};

module.exports = NoteAssocPlant;
