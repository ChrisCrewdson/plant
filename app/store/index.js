const { createStore, applyMiddleware } = require('redux');
const reducers = require('../reducers'); // combineReducers already called on reducers in her)e
const api = require('../middleware/api');
const { setupSubscribe: userSubscribe } = require('./user');
const Immutable = require('immutable');

let middleware = [api];

if (process.env.NODE_ENV !== 'production') {
  const logger = require('../middleware/logger');
  middleware = [logger, api];
}

// Add the api to the pipeline/chain
const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);

const store = createStoreWithMiddleware(reducers, Immutable.fromJS(window.__INITIAL_STATE__ || {}));

userSubscribe(store);

module.exports = store;
