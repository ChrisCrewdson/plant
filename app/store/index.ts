export {}; // To get around: Cannot redeclare block-scoped variable .ts(2451)

const { createStore, applyMiddleware } = require('redux');

const seamless = require('seamless-immutable').static;
const reducers = require('../reducers'); // combineReducers already called on reducers in her)e
const api = require('../middleware/api');
const { setupSubscribe: userSubscribe } = require('./user');

const middleware = [api];

if (process.env.NODE_ENV !== 'production') {
  // eslint-disable-next-line global-require
  const logger = require('../middleware/logger');
  middleware.unshift(logger);
}

// Add the api to the pipeline/chain
const createStoreWithMiddleware = applyMiddleware(...middleware)(createStore);

// @ts-ignore - __INITIAL_STATE__ is added on the server
const initialState = seamless.from(window.__INITIAL_STATE__ || {});
const store = createStoreWithMiddleware(reducers, initialState);

userSubscribe(store);

module.exports = store;
