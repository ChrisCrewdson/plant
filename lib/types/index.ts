
// Definitions file for Plant Project

// Currently NonEmptyList is not used anywhere.
// Plan to tag the plantIds in a Note with this as there
// always has to be at least 1 plant for each note.
type NonEmptyList<T> = T[] & { 0: T };

// Now included with Typescript 3.5+
// type Omit<T, K extends keyof T> = Pick<T, Exclude<keyof T, K>>

/*
In TypeScript, just as in ECMAScript 2015, any file containing a top-level import or export
is considered a module. Conversely, a file without any top-level import or export declarations
is treated as a script whose contents are available in the global scope (and therefore to
modules as well).
*/

type GeoType = 'Point';

interface GeoCoords {
  /** Longitude */
  0: number;
  /** Latitude */
  1: number;
}

interface Geo {
  type: GeoType; // "Point" - is this the only value I know of that's valid
  coordinates: GeoCoords;
}
interface Window {
  [key: string]: any;
  // FormData: Function;
}

type Mutable<T> = {
  -readonly [P in keyof T]: T[P];
};

interface Global extends NodeJS.Global {
  [key: string]: any;
  // window: Window;
  // requestAnimationFrame?: (callback: Function) => void;
}

type Html5InputTypes =
  | 'button'
  | 'checkbox'
  | 'color'
  | 'date'
  | 'datetime-local'
  | 'email'
  | 'file'
  | 'hidden'
  | 'image'
  | 'month'
  | 'number'
  | 'password'
  | 'radio'
  | 'range'
  | 'reset'
  | 'search'
  | 'submit'
  | 'tel'
  | 'text'
  | 'time'
  | 'url'
  | 'week';
