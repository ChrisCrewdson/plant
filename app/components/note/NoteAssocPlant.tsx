import { Dispatch } from 'redux';
import React from 'react';
import PropTypes from 'prop-types';

import Fab from '@material-ui/core/Fab';
import ArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import ArrowRight from '@material-ui/icons/KeyboardArrowRight';

import InputComboText from '../common/InputComboText';
import Errors from '../common/Errors';
import utils from '../../libs/utils';
import { actionFunc } from '../../actions';
import NoteAssocPlantToggleButton from './NoteAssocPlantToggleButton';
import { PlantAction } from '../../../lib/types/redux-payloads';

interface NoteAssocPlantProps {
  dispatch: Dispatch<PlantAction<any>>;
  error: string;
  plantIds: string[];
  plants: Record<string, UiPlantsValue>;
}

interface NoteAssocPlantState {
  expanded: boolean;
  filter: string;
}

const getSelectedState = (selected: boolean, isTerminated: boolean) => {
  if (selected) {
    return 'selected';
  }
  return isTerminated ? 'dead' : 'alive';
};

export default class NoteAssocPlant extends React.Component {
  props!: NoteAssocPlantProps;

  // eslint-disable-next-line react/state-in-constructor
  state: NoteAssocPlantState;

  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    error: PropTypes.string,
    plantIds: PropTypes.arrayOf(PropTypes.string).isRequired,
    plants: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
  };

  static defaultProps = {
    error: '',
  };

  constructor(props: NoteAssocPlantProps) {
    super(props);
    // eslint-disable-next-line react/state-in-constructor
    this.state = {
      expanded: false,
      filter: '',
    };
    this.expand = this.expand.bind(this);
    this.toggle = this.toggle.bind(this);
    this.changeHandler = this.changeHandler.bind(this);
  }

  changeHandler(_: string, value: string) {
    return this.setState({ filter: value.toLowerCase() });
  }

  /**
   * Toggles the selected state for the button for an associated plant
   */
  toggle(plantId: string) {
    const {
      plantIds: propPlantIds,
      dispatch,
    } = this.props;
    const plantIds = propPlantIds.indexOf(plantId) >= 0
      ? propPlantIds.filter((pId) => pId !== plantId)
      : propPlantIds.concat(plantId);

    dispatch(actionFunc.editNoteChange({ plantIds }));
  }

  expand() {
    const { expanded: exp } = this.state;
    const expanded = !exp;
    this.setState({ expanded });
  }

  /**
   * Renders a button that allows for attaching or removing a plant on
   * a note.
   */
  renderPlantButton(plant: UiPlantsValue, primary: boolean) {
    const { _id, title, isTerminated } = plant;
    const selectState = getSelectedState(primary, !!isTerminated);

    return (
      <NoteAssocPlantToggleButton
        _id={_id || ''}
        key={_id}
        label={title}
        selectState={selectState}
        style={{ margin: 12 }}
        toggleFunc={this.toggle}
      />
    );
  }

  renderPlantButtons(plantIds: ReadonlyArray<string>, plants: Record<string, UiPlantsValue>,
    selected: boolean) {
    return plantIds.map((plantId) => {
      const plant = plants[plantId];
      if (!plant) {
        // console.warn(`Missing plant for plantId ${plantId}`);
        return null;
      }
      return this.renderPlantButton(plant, selected);
    });
  }

  render() {
    const { expanded, filter } = this.state;
    const { plantIds, plants, error } = this.props;
    // TODO: If the following pattern is to be kept then:
    // If expanded is true then it might be faster to sort the array once before applying
    // the filter twice rather than filtering twice and sorting twice.
    const checkedPlantIds = utils.filterSortPlants(plantIds, plants, filter);
    const checkedPlants = this.renderPlantButtons(checkedPlantIds, plants, true);

    let uncheckedPlants = null;
    if (expanded) {
      const uncheckedIds = Object.keys(plants).filter(
        (plantId) => plantIds.indexOf(plantId) === -1);
      const uncheckedPlantIds = utils.filterSortPlants(uncheckedIds, plants, filter);

      uncheckedPlants = this.renderPlantButtons(uncheckedPlantIds, plants, false);
    }

    const title = `${expanded ? 'Hide' : 'Show'} Unchecked Plants`;
    const arrow = (
      <Fab
        size="medium"
        onClick={this.expand}
        color="secondary"
        title={title}
      >
        {expanded
          ? <ArrowLeft />
          : <ArrowRight />}
      </Fab>
    );

    const filterInput = (
      <InputComboText
        id="note-assoc-plant-filter"
        changeHandler={this.changeHandler}
        label="Filter"
        placeholder="Filter..."
        value={filter}
        name="filter"
      />
    );

    const errors = error ? [error] : [];

    return (
      <div style={{ textAlign: 'left' }}>
        <Errors errors={errors} />
        <div>
          {'Associated plants:'}
          {filterInput}
          {checkedPlants}
          {uncheckedPlants}
          {arrow}
        </div>
      </div>
    );
  }
}
