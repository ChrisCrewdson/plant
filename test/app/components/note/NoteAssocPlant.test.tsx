import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { Dispatch } from 'redux';
import NoteAssocPlant from '../../../../app/components/note/NoteAssocPlant';
import store from '../../../../app/store';
import App from '../../../../app/components/App';

import { PlantAction } from '../../../../lib/types/redux-payloads';

const muiTheme = getMuiTheme(lightBaseTheme);

describe('NoteAssocPlant', () => {
  const storeDispatch = store.dispatch;

  beforeAll(() => {
    store.dispatch = (() => {}) as Dispatch<PlantAction<any>>;
  });

  afterAll(() => {
    store.dispatch = storeDispatch;
  });

  test('NoteAssocPlant should be rendered', () => {
    const plantIds = ['p-1', 'p-2'];
    const plants = {
      'p-1': {
        _id: 'p-1',
        title: 'Plant One',
        isTerminated: false,
      },
      'p-2': {
        _id: 'p-2',
        title: 'Plant Two',
        isTerminated: false,
      },
      'p-3': {
        _id: 'p-3',
        title: 'Plant Three',
      },
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Provider store={store}>
          <App>
            <MemoryRouter>
              <NoteAssocPlant
                dispatch={storeDispatch}
                plantIds={plantIds}
                plants={plants}
              />
            </MemoryRouter>
          </App>
        </Provider>
      </MuiThemeProvider>);
    const badgeTotals = component.toJSON();
    expect(badgeTotals).toMatchSnapshot();
  });
});
