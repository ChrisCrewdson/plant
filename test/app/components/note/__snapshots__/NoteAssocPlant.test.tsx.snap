// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteAssocPlant NoteAssocPlant should be rendered 1`] = `
<div
  className="react-root"
>
  <div
    style={
      Object {
        "textAlign": "left",
      }
    }
  >
    <div>
      Associated plants:
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
        style={
          Object {
            "marginLeft": 20,
          }
        }
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated"
          data-shrink={false}
          htmlFor="note-assoc-plant-filter"
          id="note-assoc-plant-filter-label"
          style={
            Object {
              "fontSize": "14px",
            }
          }
        >
          Filter
        </label>
        <div
          className="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl"
          onClick={[Function]}
          style={
            Object {
              "fontSize": "14px",
            }
          }
        >
          <input
            aria-invalid={false}
            className="MuiInputBase-input MuiInput-input"
            disabled={false}
            id="note-assoc-plant-filter"
            name="filter"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            placeholder="Filter..."
            required={false}
            type="text"
            value=""
          />
        </div>
        
      </div>
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "margin": 12,
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#00bcd4",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "margin": 0,
                    "muiPrepared": true,
                    "opacity": 1,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "userSelect": "none",
                  }
                }
              >
                Plant One
              </span>
            </div>
          </div>
        </button>
      </div>
      <div
        style={
          Object {
            "WebkitTapHighlightColor": "rgba(0,0,0,0)",
            "backgroundColor": "#ffffff",
            "borderRadius": 2,
            "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12)",
            "boxSizing": "border-box",
            "color": "rgba(0, 0, 0, 0.87)",
            "display": "inline-block",
            "fontFamily": "Roboto, sans-serif",
            "margin": 12,
            "minWidth": 88,
            "muiPrepared": true,
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          }
        }
      >
        <button
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
              "backgroundColor": "#00bcd4",
              "border": 10,
              "borderRadius": 2,
              "boxSizing": "border-box",
              "cursor": "pointer",
              "display": "inline-block",
              "fontFamily": "Roboto, sans-serif",
              "fontSize": "inherit",
              "fontWeight": "inherit",
              "height": 36,
              "lineHeight": "36px",
              "margin": 0,
              "muiPrepared": true,
              "outline": "none",
              "padding": 0,
              "position": "relative",
              "textAlign": "center",
              "textDecoration": "none",
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "verticalAlign": null,
              "width": "100%",
            }
          }
          tabIndex={0}
          type="button"
        >
          <div
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchStart={[Function]}
          >
            <div
              style={
                Object {
                  "backgroundColor": false,
                  "borderRadius": 2,
                  "height": 36,
                  "muiPrepared": true,
                  "top": 0,
                  "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
                }
              }
            >
              <span
                style={
                  Object {
                    "color": "#ffffff",
                    "fontSize": 14,
                    "fontWeight": 500,
                    "letterSpacing": 0,
                    "margin": 0,
                    "muiPrepared": true,
                    "opacity": 1,
                    "paddingLeft": 16,
                    "paddingRight": 16,
                    "position": "relative",
                    "textTransform": "uppercase",
                    "userSelect": "none",
                  }
                }
              >
                Plant Two
              </span>
            </div>
          </div>
        </button>
      </div>
      <button
        className="MuiButtonBase-root MuiFab-root MuiFab-sizeMedium MuiFab-secondary"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        title="Show Unchecked Plants"
        type="button"
      >
        <span
          className="MuiFab-label"
        >
          <svg
            aria-hidden="true"
            className="MuiSvgIcon-root"
            focusable="false"
            role="presentation"
            style={
              Object {
                "fontSize": "3em",
              }
            }
            viewBox="0 0 24 24"
          >
            <path
              d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
